File: frontend/src/app/services/ai-recommendation.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {
  AiRecommendationRequestDto,
  ApiResponse,
  RecommendationDto,
} from '../models/models';
import { environment } from '../../environments/environment';

const API_BASE = environment.apiUrl;

@Injectable({ providedIn: 'root' })
export class AiRecommendationService {
  private readonly http = inject(HttpClient);

  recommend(
    request: AiRecommendationRequestDto
  ): Observable<ApiResponse<RecommendationDto>> {
    return this.http.post<ApiResponse<RecommendationDto>>(
      `${API_BASE}/ai/recommendations`,
      request
    );
  }
}

File: backend/src/main/java/com/yaquedo/ai/service/AiRecommendationService.java
package com.yaquedo.ai.service;

import com.yaquedo.ai.dto.AiRecommendationRequestDto;
import com.yaquedo.dto.RecommendationDto;

public interface AiRecommendationService {

    RecommendationDto recommend(AiRecommendationRequestDto request, String userEmail);
}
